File: 4.8-bc/ts/stage-3-hexagonal/src/auth/infra/http/dto.ts
export interface RegisterReq {
  email?: string;
  password?: string;
}

export interface RegisterResp {
  user_id: string;
  email: string;
  created_at: Date;
}

export interface LoginReq {
  email?: string;
  password?: string;
}

export interface LoginResp {
  user_id: string;
  email: string;
}

File: 4.8-bc/py/stage-3-hexagonal/app/auth/infra/http/dto.py
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, EmailStr


class RegisterReq(BaseModel):
    email: EmailStr
    password: str


class RegisterResp(BaseModel):
    user_id: str
    email: str
    created_at: datetime


class LoginReq(BaseModel):
    email: EmailStr
    password: str


class LoginResp(BaseModel):
    user_id: str
    email: str
